src/main/scala/org/clulab/habitus/actions/HabitusActions.scala

```diff
@@ -186,20 +207,95 @@ class HabitusActions extends Actions {
+    "Quantity" -> Set("t/ha", "kg/ha", "kg", "d", "cm", "mg/l", "kg n ha-1"),
+    "AreaSize" -> Set("ha"),
+    "YieldAmount" -> Set("t/ha", "kg/ha", "kg"),
+    "FertilizerQuantity" -> Set("kg/ha", "mg/l", "kg n ha-1")
   )
 
+  def hasLetters(string: String): Boolean = {
+    string.exists(ch => ch.isLetter || ch.toString == "/")
+  }
+
+  def getNormString(m: Mention): String = {
+    m.norms.get.head.split("\\s").filter(hasLetters).mkString(" ") // e.g., get "m2" from "6 m2"
+  }
+
+
   def measurementIsAppropriate(m: Mention): Boolean = {
+    // check mention type
+    val probableUnit = m match {
+      case tbm: TextBoundMention => getNormString(tbm)
+      case rm: RelationMention => getNormString(rm.arguments("value").head)
+      case em: EventMention => getNormString(em.arguments("value").head)
+      case _ => throw new RuntimeException(s"Unknown mention type ${m.getClass}")
+    }
     // check if any of the possible units shows up in the norm
-    val probablyUnit = m.norms.get.head.split("\\s").last // e.g., get "m2" from "6 m2"; assume value is separated from unit with a space and unit is one token
     labelToAppropriateUnits
-      .getOrElse(m.label, throw new RuntimeException(s"Unknown measurement label ${m.label}"))(probablyUnit)
+      .getOrElse(m.label, throw new RuntimeException(s"Unknown measurement label ${m.label}"))(probableUnit)
   }
 
   def appropriateMeasurement(mentions: Seq[Mention]): Seq[Mention] = {
     // applies to individual rules to check if the measurement is appropriate for the mention label
     mentions.filter(measurementIsAppropriate)
   }
 
+  def makeEventFromUnitSplitByFertilizer(m: Mention): Mention = {
+    val fertilizer = m.arguments("fertilizer").head
+    val value = m.arguments("number").map(_.text)
+    val unit1 = m.arguments("unit1").map(_.text)
+    val unit2 = m.arguments("unit2").head.text.replace("-1", "")
+    val unit = unit1.mkString("") + "/" + unit2
+    val newMention = new MeasurementMention(
+      m.labels,
+      m.tokenInterval,
+      m.sentence,
+      m.document,
+      m.keep,
+      m.foundBy,
+      m.attachments,
+      Some(value),
+      Some(Seq(unit)),
+      false
+    )
+    val newArgs = Map("variable" -> Seq(fertilizer), "value" -> Seq(newMention))
+    val newEvent = new RelationMention(
+      labels = Seq("FertilizerQuantity", "Event"),
+      tokenInterval = m.tokenInterval,
+      arguments = newArgs,
+      paths = m.paths,
+      sentence = m.sentence,
+      document = m.document,
+      keep = m.keep,
+      foundBy = m.foundBy + "++makeQuantityFromUnitSplitByFertilizer",
+      attachments = m.attachments
+    )
+    newEvent
+  }
+
+  def adjustQuantityNorm(mentions: Seq[Mention]): Seq[Mention] = {
+    mentions.map { m =>
+      val value1 = m.arguments("value1").head.text.toFloat
+      val value2 = m.arguments("value2").head.text.toFloat
+      val unit = m.arguments("unit")
+      val valueRange = s"${value1} -- ${value2}"
```

Comment:
There is a danger here that you won't get the expected string.  For example,
```scala
scala> val f1 = "0.0001".toFloat
f1: Float = 1.0E-4

scala> val f1 = "33333.333".toFloat
f1: Float = 33333.332
```

Reply:
Keith please do you have any suggestion for this?

Reply:
It might be even worse, but one could use the original text: `m.arguments("value1").head.text`.  If someone wrote "3.14", "03.14", "0.314E1", "+3.14" or "--3.1.4" it might be a problem that there is more than one representation of the same value.  To avoid the rounding error one could start with .toDouble or (better) use a BigDecimal, and to keep from switching to scientific notation one can probably use f"" with some kind of formatting instead of s"".  (https://docs.scala-lang.org/overviews/core/string-interpolation.html).  I don't know how important it is to account for this, but did want to flag potential problems.

Reply:
I'd be perfectly happy if a comment was added near those lines so that the next person or our future selves are alerted to the possible gotcha.